DataProcessing/CoinGeckoDataDownloader.cs

```diff
@@ -0,0 +1,383 @@
+        {
+            var coins = new HashSet<string>();
+
+            var symbolPropertiesDatabase = SymbolPropertiesDatabase.FromDataFolder();
+
+            var fiat = symbolPropertiesDatabase.GetSymbolPropertiesList(Market.Oanda, SecurityType.Forex)
+                .Select(x => x.Value.QuoteCurrency.ToLowerInvariant()).ToHashSet();
+
+            foreach (var symbolPropertiesFromMarket in Market.SupportedMarkets()
+                         .Select(market => symbolPropertiesDatabase.GetSymbolPropertiesList(market, SecurityType.Crypto)))
+            {
+                foreach (var (key, symbolProperties) in symbolPropertiesFromMarket)
+                {
+                    var coin = key.Symbol.ToLowerInvariant();
+                    var quote = symbolProperties.QuoteCurrency.ToLowerInvariant();
+                    var end = coin.Length - quote.Length;
+                    coin = coin[..end];
+
+                    if (!fiat.Contains(coin)) coins.Add(coin);
+                    if (!fiat.Contains(quote)) coins.Add(quote);
+                }
+            }
+
+            return coins;
+        }
+
+        /// <summary>
+        /// Get the list of coins supported by CoinGecko
+        /// This method saves the data locally since we might need to rerun
+        /// this data downloader and the rate limit of CoinGecko API is high 
+        /// </summary>
+        /// <returns>List of coins supported by CoinGecko</returns>
+        private List<CoinGeckoItem> GetCoinGeckoList()
+        {
+            string value;
+
+            if (File.Exists("list.json"))
+            {
+                value = File.ReadAllText("list.json");
+            }
+            else
+            {
+                value = HttpRequester("list").Result;
+                File.WriteAllText("list.json", value);
+            }
+
+            // Remove Wormhole duplicates
+            var coinGeckoList = JsonConvert.DeserializeObject<List<CoinGeckoItem>>(value)
+                .GroupBy(x => x.Symbol)
+                .SelectMany(x => x.Where(y => !y.Name.ToLowerInvariant().Contains("wormhole")))
+                .ToList();
+
+            return coinGeckoList;
+        }
+
+        /// <summary>
+        /// Gets the data for a given coin Id
+        /// This method saves the data locally since we might need to rerun
+        /// this data downloader and the rate limit of CoinGecko API is high 
+        /// </summary>
+        /// <param name="id">Coin Id</param>
+        /// <param name="exists">is the processed file exists</param>
+        /// <returns>Data from the MarketCharts endpoint for a given Id</returns>
+        private CoinGeckoMarketChart GetCoinGeckoMarketChartsForId(string id, bool exists)
+        {
+            string value;
+            if (File.Exists($"{id}.json"))
+            {
+                value = File.ReadAllText($"{id}.json");
+            }
+            else
+            {
+                var days = exists ? "1" : "max";
```

Comment:
Is `max` enough to cover the whole history? for example for BTC, would be nice to add a comment of what max is

Reply:
Seems it is, BTC data is complete 👍